### PR TITLE
add supportPython3 option to plugin metadata, and disable incompatible plugins

### PR DESCRIPTION
**Pull request by 3nids**
Here is an idea on how to support existing plugins while running QGIS built with Qt5/Py3.

The idea is to have a new tag that tells if the plugin supports Python 3 or not, which would be false by default.

I suppose this would require the update of the plugin repository website code, but I don't have access to it.

Any opinion @m-kuhn @borysiasty @NathanW2 ?

### CODE REVIEW

**Comment by elpaso** [comment]
I have access to the repo code, if you make a PR I can merge it, or just let me know if I have to introduce the new metadata boolean tag.

Please note that this new tag would require updates in the documentation (cookbook etc.).

**Comment by m-kuhn** [comment]
Thanks a lot for this!

Will we assume that every plugin supports also python 2? I think after releasing a py3 release we will have some py3 only plugins which should not be listed as supported in the last living LTR installations.

**Comment by 3nids** [comment]
@elpaso the idea is to first propose the idea. Let's see, and later I can PR, but where is the repo for this, I didn't find anything on github???

@m-kuhn indeed, this aims at be reverting whenever we start QGIS 3. For now, we assume all plugins are py2 only by default and py3 compatible if specified witht this option. Later on, all plugins are py3, without any info regarding py2.

**Comment by elpaso** [comment]
The repo is here: https://github.com/qgis/QGIS-Django

**Comment by 3nids** [comment]
ok, here we go https://github.com/qgis/QGIS-Django/pull/20 is this ok?

**Comment by 3nids** [comment]
@m-kuhn I'm changing my mind....shouldn't we be able to support py3 only too?

**Comment by NathanW2** [comment]
+1 for a filter like this.  Even if we ignore it later on we still need it
for the time being.

**Comment by 3nids** [comment]
ok, so we would add two keys: supportPyQt4 (default true) and supportPyQt5 (default false) ?

**Comment by NathanW2** [comment]
If we are going with PyQt4 == Python 2 and PyQt5 == Python 3 yes.

**Comment by m-kuhn** [comment]
> If we are going with PyQt4 == Python 2 and PyQt5 == Python 3 yes.

I'd say we don't need to enforce this in cmake/cpp, but we can simplify the possibility matrix to these two for the plugin repo I guess.

**Comment by 3nids** [comment]
now python can handle either one python version or both, still assuming py3 = qt5 and py2 = qt4.
@m-kuhn thoughts?

File: python/pyplugin_installer/installer_data.py [hunk 1]
```diff
@@ -450,6 +450,19 @@ def xmlDownloaded(self):
                         "library": "",
                         "readonly": False
                     }
+
+					supportPython2 = pluginNodes.item(i).firstChildElement("supportPython2").text().strip()
```
Comment (m-kuhn) [inline python/pyplugin_installer/installer_data.py]: Looks like you have used tabs instead of spaces.

**Pull request closed without merging**